### ISSUE TITLE
linter: `ignore-pattern` doesn't seem to work properly

### ISSUE THREAD

**Issue by Lutymane** [oxc-project/oxc#7102]
### What version of Oxlint are you using?

0.11.0

### What command did you run?

pnpm dlx oxlint@latest --ignore-pattern="./ignore/**"

### What does your `.oxlint.json` config file look like?

No config

### What happened?

`ignore-pattern` argument doesn't seem to work at all. I tested this simple setup:

```js
// index.js

let a = new Array(2);
```

```js
// ignore/index.js

let a = new Array(2);
```

oxlint still processes both files, although it should skip `ignore/index.js` one. I tried all kinds of arguments:
* `".\\ignore\\**"`
* "./ignore/**/*.js"`
* and etc

What's interesting it works if I just specify directory `"ignore"`

But then I add file `ignore/nested/index.js` and nothing works at all to ignore that file not even similar style directory only argument `"ignore/nested"`, so there's no way to skip nested folders as it seems

I'm using Windows

**Comment by camc314** [comment]
possibly fixed by https://github.com/oxc-project/oxc/pull/7407 ?

**Comment by 1zumii** [comment]
@Lutymane

### TLDR
do not use `./` in glob syntax, and ignore pattern works as it specified

* * *

it seems not a bug, i tried with the [demo](https://gitee.com/xxx8848/oxlint-7102) according to you listed above:
```rust
.
├── ignore
│   ├── another.jsx  // contains problems
│   └── index.js     // contains problems
├── index.js         // contains problems
└── package.json
```
- run linter with `--ignore-pattern=./ignore/**`
  all files report lint problems, includes those under `ignore/`.
  - [x] index.js
  - [x] ignore/index.js
  - [x] ignore/another.jsx

- run linter with `--ignore-pattern=ignore/**` (striped leading `./`)
  only `index.js` reports, problems from `ignore/` files are correctly ignored.
  - [x] index.js
  - [ ] ignore/index.js
  - [ ] ignore/another.jsx

- run linter with `--ignore-pattern=ignore/**.jsx`
  `ignore/another.jsx` problems are ignored, `index.js` and `ignore/index.js` report.
  - [x] index.js
  - [x] ignore/index.js
  - [ ] ignore/another.jsx
